### PR TITLE
[serdepp] add new port

### PR DESCRIPTION
**Pull request by injae**
**Describe the pull request**

- #### What does your PR fix?  
   Adds a [serdepp](https://github.com/injae/serdepp) port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by injae** [comment]
It doesn't seem to find the files installed with ExternalProject_add.
---

x64_linux error.log
```
CMake Error at cmake/cppm_loader.cmake:30 (include):
  include could not find requested file:

    /Users/vagrant/.cppm/cmake/cppm-loader/load.cmake
Call Stack (most recent call first):
  CMakeLists.txt:4 (include)


CMake Error at CMakeLists.txt:5 (cppm_project):
  Unknown CMake command "cppm_project".
```

cppm-loader/load.cmake
```
if(CMAKE_SYSTEM_NAME STREQUAL "Windows" OR (NOT CMAKE_SYSTEM_NAME AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows"))
    set(env_home "$ENV{USERPROFILE}")
else()
    set(env_home "$ENV{HOME}")
endif()
string(REPLACE "\\" "/" HOME "${env_home}")

set(CPPM_ROOT          ${HOME}/.cppm)
set(CPPM_LOADER_CACHE  ${CPPM_ROOT}/cache/cppm-loader/git)
set(CPPM_LOADER_PATH   ${CPPM_ROOT}/cmake/cppm-loader)

if(NOT DEFINED IS_CPPM_LOADER_LOADED)
set(_install_script "${CPPM_LOADER_CACHE}/install-script")
file(WRITE ${_install_script}/CMakeLists.txt
"cmake_minimum_required(VERSION 3.6)
project(CPPM_LOADER_DOWNLOAD NONE)
include(ExternalProject)
ExternalProject_Add(cppm-loader
    GIT_REPOSITORY https://github.com/injae/cppm-loader.git
    SOURCE_DIR ${CPPM_LOADER_PATH}
    BINARY_DIR ${CPPM_LOADER_CACHE}/build
    CONFIGURE_COMMAND \"\"
    BUILD_COMMAND  \"\"
    INSTALL_COMMAND \"\"
)"
)
execute_process(COMMAND cmake . WORKING_DIRECTORY ${_install_script} OUTPUT_QUIET)
execute_process(COMMAND cmake  --build . WORKING_DIRECTORY ${_install_script} OUTPUT_QUIET)
endif()
include(${CPPM_LOADER_PATH}/load.cmake)
```

**Comment by JonLiu1993** [comment]
@injae ,Thanks for your pr, this is the detail error log:
https://dev.azure.com/vcpkg/public/_build/results?buildId=65686&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by injae** [comment]
ExternalProject_Add does not seem to work because of the problem that find_package(GIT) does not work in portfile.cmake.

```  
error: could not find git for clone of cppm-loader
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
  CMakeLists.txt:4 (ExternalProject_Add)
```

File: ports/serdepp/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "serdepp",
+  "version-string": "0.1.4",
```
Comment (JonLiu1993) [inline ports/serdepp/vcpkg.json:3]:
```suggestion
  "version": "0.1.4",
```

File: ports/serdepp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/serdepp)
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/cmake
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/lib/cmake
+)
+
+# # Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/serdepp RENAME copyright)
```
Comment (JonLiu1993) [inline ports/serdepp/portfile.cmake:28]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

File: ports/serdepp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO injae/serdepp
+    REF v0.1.4
+    SHA512 da84ad82e882c0cada5c9dd3c77afd45aaf7b3b4eb150257b09b9b4854b349fdb2c39be2f6ba40bb39b34262e44609a02afba1ec860625f25a2313f7ac34a055
+    HEAD_REF main
+)
+
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/serdepp/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/serdepp/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/serdepp)
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/cmake
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/lib/cmake
+)
```
Comment (JonLiu1993) [inline ports/serdepp/portfile.cmake:25]:
```suggestion
file(REMOVE_RECURSE
    "${CURRENT_PACKAGES_DIR}/debug/cmake"
    "${CURRENT_PACKAGES_DIR}/debug/include"
    "${CURRENT_PACKAGES_DIR}/lib/cmake"
)
```

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**